src/main/java/io/vertx/core/parsetools/impl/LengthFieldParserImpl.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2011-2021 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.core.parsetools.impl;
+
+import io.vertx.core.Handler;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.impl.Arguments;
+import io.vertx.core.parsetools.LengthFieldParser;
+import io.vertx.core.parsetools.RecordParser;
+import io.vertx.core.streams.ReadStream;
+
+/**
+ * @author <a href="mailto:emad.albloushi@gmail.com">Emad Alblueshi</a>
+ */
+
+public class LengthFieldParserImpl implements LengthFieldParser {
+
+  private Handler<Throwable> exceptionHandler;
+  private final RecordParser parser;
+
+  private long frameLength = -1;
+  private final int length;
+  private final int offset;
+  private final int max;
+
+  public LengthFieldParserImpl(int length, int offset, int max, ReadStream<Buffer> stream) {
+    Arguments.require(length == 1 || length == 2 || length == 3 || length == 4 || length == 8,
+      "Field length must be 1, 2, 3, 4, or 8");
+    Arguments.require(offset >= 0, "Field offset must be >= 0");
+    Arguments.require(max > 0, "Max frame length must be > 0");
+    this.length = length;
+    this.offset = offset;
+    this.max = max;
+    this.parser = RecordParser.newFixed(length + offset, stream);
+  }
+
+  @Override
+  public LengthFieldParser exceptionHandler(Handler<Throwable> handler) {
+    if (handler != null) {
+      exceptionHandler = handler;
+    }
+    return this;
+  }
+
+  @Override
+  public LengthFieldParser handler(Handler<Buffer> handler) {
+    if (handler == null) {
+      parser.handler(null);
+      parser.exceptionHandler(null);
+      parser.endHandler(null);
+      return this;
+    }
+    parser.handler(buffer -> {
+      try {
+        if (frameLength == -1) {
+          frameLength = handleFrameLength(buffer);
+          if (frameLength > max || frameLength <= 0) {
+            try {
+              String err = frameLength <= 0 ? "Frame length : " + frameLength + " <= 0" :
+                "Frame length is too large current: " + frameLength + " max: " + max;
+              IllegalStateException ex = new IllegalStateException(err);
+              if (exceptionHandler != null) {
+                exceptionHandler.handle(ex);
+              } else {
+                throw ex;
+              }
+            } finally {
+              frameLength = 0;
```

Comment:
It would be better to set the `RecordParser` handlers to `null`. This will ensure the underlying stream is closed eagerly.

Reply:
The parser won't work correctly in case of null stream is provided in the constructor :-) we should keep in mind that the current parser and `RecordParser` handle both cases.

1) If stream is null the exception is thrown and **keep reading**. 
2) If stream is not null an exception is thrown and **stop reading** then the user can handle the exception and close the stream explicitly.

I discovered this after heavy testing for the above two cases using `RecordParser` alone with fixed and delimited modes!

So the `frameLength = 0`  covers both cases accordingly.

Reply:
When you create a `Flowable` from a `ReadStream` with `vertx-rx`, we need a way to close the stream if the subscription is canceled. This is done by setting handlers to `null`.

So this method must be prepared to receive a `null` handler.

Reply:
I think I should add `this.handler(null)`